### PR TITLE
Return an error if a document _id exists on insertion 

### LINKED ISSUE #143: Return an error if a document _id exists on insertion
See #142 

We should return an error if an insert tries to insert a document with an existing _id

### PR DESCRIPTION
**Pull request by maheshrajamani**
**What this PR does**:
Validate if inserted _id is unique.

**Which issue(s) this PR fixes**:
Fixes #143 

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [ ] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/InsertIntegrationTest.java [hunk 1]
```diff
@@ -112,6 +112,71 @@ public void insertDocumentWithNumberId() {
...
+          .statusCode(200)
+          .body("data.docs[0]", jsonEquals(expected));
+
+      json =
+          """
+                {
+                  "insertOne": {
+                    "document": {
+                      "_id": "duplicate",
+                      "username": "user4"
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/InsertIntegrationTest.java:164]: Maybe try inserting document with different `username`, and after failure make sure nothing was changed?
(i.e. no Upsert happened)

**Review by tatu-at-datastax — APPROVED** [review]
Looks good: may want to add verification that failing insert actually made no change -- that is, it doesn't just report failure but also avoids doing upsert.

**Comment by jeffreyscarpenter** [comment]
So the JSON API will not support upsert semantics?

**Comment by maheshrajamani** [comment]
Upsert will be part of findOneAndUpdate and Update command, not part of Insert.

**Review by vkarpov15 — COMMENTED** [review]
Nice :+1:

**Pull request merged**